src/ml_gym/persistency/io.py

```diff
@@ -35,34 +49,103 @@ def get_experiment_statuses(self, grid_search_id: str) -> List[ExperimentStatus]
 
 
 class GridSearchRestfulAPIClient(GridSearchAPIClientIF):
+    """
+    GridSearchRestfulAPIClient class
+
+    Contains the function implementation from GridSearchAPIClientIF for HTTP calls to be made.
+    :params: GridSearchAPIClientIF
+    """
 
     def __init__(self, endpoint: str):
         self.endpoint = endpoint
 
     def _get_json_resource(url) -> Dict:
+        """
+        ``HTTP GET Call`` Fetch json data.
+
+        :params: 
+             endpoint (str): HTTP request URL
+
+        :returns: bytes data
+        """
         response = requests.get(url)
         if response.status_code != HTTPStatus.OK:
-            raise NetworkError(f"Server responded with error code {response.status_code}")
+            raise NetworkError(
+                f"Server responded with error code {response.status_code}")
         else:
             try:
                 grid_search_config = response.json()
             except ValueError as e:
-                raise DataIntegrityError(f"Could not cast response from {url} to json.") from e
+                raise DataIntegrityError(
+                    f"Could not cast response from {url} to json.") from e
             return grid_search_config
 
     def _get_binary_resource(url) -> bytes:
+        """
+        ``HTTP GET Call`` Fetch bytes data.
+
+        :params: 
+             url (str): HTTP request URL
+
+        :returns: bytes data
+        """
         response = requests.get(url)
         if response.status_code != HTTPStatus.OK:
-            raise NetworkError(f"Server responded with error code {response.status_code}")
+            raise NetworkError(
+                f"Server responded with error code {response.status_code}")
         else:
             return response.content
 
+    def _set_binary_resource(url: str, payload: bytes) -> Dict:
+        """
+        ``HTTP POST Call`` Send bytes data.
+
+        :params: 
+             url (str): HTTP request URL
+             payload (bytes); pickle dump to be sent
+        """
+        data = base64.b64encode(payload)
+        response = requests.post(url=url, data={
+                                 'data': data, "msg": "insert checkpoint", "type": "multipart/form-data"}, files={"file": data})
+        if response.status_code != HTTPStatus.OK:
+            raise NetworkError(
+                f"Server responded with error code {response.status_code}")
+    
+    def _del_binary_resource(url: str) -> Dict:
+        """
+        ``HTTP DELETE Call`` Delete data.
+
+        :params: 
+             url (str): HTTP request URL
+        """
+        response = requests.delete(url=url)
+        if response.status_code != HTTPStatus.OK:
+            raise NetworkError(
+                f"Server responded with error code {response.status_code}")
+
     def _set_raw_text_file_resource(url: str, payload: Dict) -> Dict:
```

Comment:
would rename this to `_put_raw_text_file_resource`

Reply:
Done